script/2_data_wrangling/wrangle_hazard.R

```diff
@@ -62,3 +68,226 @@ summary(nc_grid$sems_impact)
 
 # WRITE OUT RESULTS 
 st_write(nc_grid, "data/output/nc_grid_hazard.gpkg", layer = "hazard", delete_layer = TRUE)
+
+### EXTRACT IMPACTS FROM CAFO LOCATIONS
+
+## 1. nc_grid is already in memory
+
+## 2. read in point sources
+cafo_df <- read_excel("data/input/CAFOs/List_OfPermitted_Animal_Facilities20241104.xlsx", sheet = 1) %>%
+  janitor::clean_names()  # convert column names to snake casa using the janitor package
+
+# Remove rows with missing lat/long
+cafo_df_clean <- cafo_df %>%
+  filter(!is.na(location_lat_num) & !is.na(location_long_num))
+
+# Convert to spatial points (assuming WGS84 / EPSG:4326)
+cafo_sf <- st_as_sf(cafo_df_clean, coords = c("location_long_num", "location_lat_num"), crs = 4326)
+
+## 3. Compute CAFO impact score
+# take a sample to test the code
+# nc_grid_sample <- nc_grid %>%
+# slice_sample(n = 5000)
+
+handlers("txtprogressbar")
+with_progress({
+  nc_grid <- compute_impact_score(points_sf = cafo_sf,
+                                  grid_sf = nc_grid,
+                                  threshold_m = 5000,
+                                  weight_var = NULL,
+                                  output_var = "cafo_impact")
+})
+
+# full run takes 1504 seconds
+# check results
+summary(nc_grid$cafo_impact)
+
+## 4. WRITE OUT RESULTS
+st_write(nc_grid, "data/output/nc_grid_hazard.gpkg", layer = "hazard", delete_layer = TRUE)
+
+### EXTRACT IMPACTS FROM TRI WATER RELEASES
+
+## 1. nc_grid is already in memory
+
+## 2. read in point sources
+tri_df <- readxl::read_excel("data/input/TRI impact/TRI Toxic Tracker_2019 - 2023.xlsx", sheet = 1) %>%
+  janitor::clean_names()  # convert column names to snake casa using the janitor package
+
+# Filter out rows missing coordinates
+tri_df_clean <- tri_df %>%
+  filter(!is.na(longitude), !is.na(latitude))
+
+# Convert to spatial points (assuming WGS84 / EPSG:4326)
+tri_sf <- st_as_sf(tri_df_clean, coords = c("longitude", "latitude"), crs = 4326)
+
+## 3. Compute impact score with water release as weight (lbs)
+# take a sample to test the code
+# nc_grid_sample <- nc_grid %>%
+# slice_sample(n = 5000)
+
+handlers("txtprogressbar")
+with_progress({
+  nc_grid <- compute_impact_score(
+    points_sf = tri_sf,
+    grid_sf = nc_grid,
+    threshold_m = 5000,
+    weight_var = "water_releases_lb",
+    output_var = "TRI.water.releases"
+  )
+})
+
+# full run takes 2177 seconds
+# check result
+summary(nc_grid$TRI.water.releases)
+
+## 4. WRITE OUT RESULTS
+st_write(nc_grid, "data/output/nc_grid_hazard.gpkg", layer = "hazard", delete_layer = TRUE)
```

Comment:
You do not need the write out code in between variables, this will overwrite the previous result.